react/week3/todo-app/src/components/TodoItem.js

```diff
@@ -0,0 +1,53 @@
+import React, { useState } from 'react';
+import { Modal } from './Modal';
+
+export function TodoItem(props) {
+  const { id, description, deadline, deleteCallBack, editCallback } = props;
+  const [checked, setChecked] = useState(false);
+  const [onEdit, setOnEdit] = useState(false);
+  const [inputText, setInputText] = useState(description);
+  const updateHandle = id => {
+    setOnEdit(false);
+    if (inputText) {
+      editCallback(inputText, id);
+    } else {
+      setInputText(description);
+    }
+  };
+
+  return (
+    <div>
+      <Modal>
+        {onEdit ? (
+          <div>
+            <input
+              type="text"
+              value={inputText}
+              onChange={e => setInputText(e.target.value)}
+            />
+            <input type="checkbox" onChange={() => setChecked(!checked)} />
+            <button onClick={() => deleteCallBack()}>Delete</button>
```

Comment:
Its kind of repetitive code here on line number 29 and 44 , I would suggest set the operational ? condition only where you need to change set of data , not on whole block, like only edit button need to change in update.